### PR TITLE
Remove hosthook api

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
cc @luqunl @jkotas

### CODE REVIEW

File: src/vm/comdelegate.cpp [hunk 1]
```diff
@@ -1433,15 +1433,12 @@ OBJECTREF COMDelegate::ConvertToDelegate(LPVOID pCallback, MethodTable* pMT)
         MethodDesc *pStubMD = pClass->m_pForwardStubMD;
         _ASSERTE(pStubMD != NULL && pStubMD->IsILStub());
 
-
-#ifdef MDA_SUPPORTED
+#if defined(MDA_SUPPORTED)
```
Comment (jkotas) [inline src/vm/comdelegate.cpp]: Is this intentional change?

File: src/vm/dllimport.cpp [hunk 2]
```diff
@@ -5592,33 +5561,13 @@ VOID NDirectMethodDesc::SetNDirectTarget(LPVOID pTarget)
...
-    }
-
 #ifdef _TARGET_X86_
 
 
 #ifdef MDA_SUPPORTED
     if (!IsQCall() && MDA_GET_ASSISTANT(PInvokeStackImbalance))
     {
-        pInterceptStub = GenerateStubForMDA(pTarget, pInterceptStub, fHook);
+        pInterceptStub = GenerateStubForMDA(pTarget, pInterceptStub, /* fCalledByStub */ FALSE);
```
Comment (jkotas) [inline src/vm/dllimport.cpp]: Delete the argument instead?

File: src/vm/comdelegate.cpp (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/vm/comdelegate.cpp]: Yep - just so it was consistent with the enclosing one for `_TARGET_X86_`

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test Ubuntu x64 Formatting

**Pull request merged**